src/datasets/types.ts

```diff
@@ -0,0 +1,50 @@
+import { DataCube } from 'src/models/data-cube/data-cube.model';
+
+export interface Dataset {
+  charts: ChartMetaType[];
+  dataCube: DataCube;
+}
+
+export type ChartMetaType = LineChartMeta | TabbedChartsMeta;
+
+export type XYChartData = LineChartData;
+
+export interface Datum<T, U> {
+  x: T;
+  y: U;
+}
+
+export interface ChartMeta<Type, DataGeneratorType> {
+  type: Type;
+  fetchs: DataGeneratorType;
+  title: string;
+  xlabel?: string;
+  ylabel?: string;
+}
+
+export interface TabbedChartsMeta {
+  type: 'tabbed';
+  charts: ChartMetaType[];
+  title: string;
+}
+
+export type LineChartMeta = ChartMeta<'line', LineChartDataGenerator>;
+
+export type LineChartDataGenerator = (options: LineChartQueryOptions) => LineChartData[];
+
+export interface LineChartQueryOptions {
+  range: [Date, Date];
+}
+
+export interface LineChartData {
+  data: Datum<Date, number>[];

```

Comment:
`data` property in `Data` sounds confusing. How does renaming `LineChartData` to `LineChartLegendItem` sound?

Reply:
How about renaming `Datum` to `XYPoint` and `LineChartData.data` to `LineChartData.points`?

Reply:
Yeah that sounds good too 👍